frontend/src/metabase-types/api/dataset.ts

```diff
@@ -0,0 +1,23 @@
+import { DownloadPermission } from "./permissions";
+
+export interface ResultsMetadata {
+  download_results?: DownloadPermission;
```

Comment:
@noahmoss what do you think on adding this prop to the `results_metadata` of `/api/dataset` handler?

Reply:
I'm not sure it makes sense nested like that. Could the BE just return it as one of the top-level fields in the response?

And this will also need to be returned by `POST /api/card/:card-id/query` I assume, right?

Reply:
To handle unsaved questions with which users add/remove tables while editing them, we can't use `/api/card/:card-id/query` because there is no card yet. Because of that I was thinking about `/api/dataset`. Would it be a problem?
Regarding the exact format sure, you can add it in any reasonable place in JSON

Reply:
Won't it be needed for both `/api/card/:card-id/query` and `/api/dataset`, to handle both saved and unsaved questions? Or am I misunderstanding?

Reply:
You're right, I misunderstood it. For saved questions we use `/api/card/:card-id/query`